File: src/tools/index_status.py
from src.types.api import success


def handle_index_status() -> dict:
    return success(
        {
            "message": "Этап 1: заглушка index_status",
            "is_indexed": False,
            "files_count": 0,
            "chunks_count": 0,
            "last_indexed_at": None,
            "stub": True,
        }
    )

File: src/tools/index_status.ts
import path from "node:path";

import { getSettings } from "../config/settings.js";
import { readIndexStatus } from "../indexing/indexer.js";
import { success, type ToolResponse } from "../types/api.js";

export const handleIndexStatus = (): ToolResponse<Record<string, unknown>> => {
  const settings = getSettings();
  const dataDir = path.dirname(path.resolve(settings.historyDbPath));
  const status = readIndexStatus(dataDir);

  return success({
    message: "Статус индекса",
    ...status,
    stub: false,
  });
};
